### PR TITLE
grub_test: Handle live menu layout to boot on hard disk

### PR DESCRIPTION
**Pull request by ggardet**
- Related ticket: https://progress.opensuse.org/issues/56285
- Verification run: https://openqa.opensuse.org/t1023229

### CODE REVIEW

**Comment by okurz** [comment]
+1

**Comment by ggardet** [comment]
Not sure what the problem is pointed by CI.

**Comment by SergioAtSUSE** [comment]
Use of ?PATTERN? without explicit operator is deprecated at tests/installation/grub_test.pm line 43.

**Comment by ggardet** [comment]
> Use of ?PATTERN? without explicit operator is deprecated at tests/installation/grub_test.pm line 43.

Yes, I read it. But what does it mean? What is wrong? There is a `=` operator.

**Comment by SergioAtSUSE** [comment]
I would say that perl and being context interpreted. It seems the parser thinks that the ? character is used as pattern instead of ternary operator.

I was trying to reproduce in my local environment, but, I don't have it configure properly to work with `make test`.
I have created a PR to have results from travis: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/8352

**Comment by SergioAtSUSE** [comment]
@ggardet, so, add () to is_livecd and that perl linter has enough information to know that what comes  is logical expression and not a bareword.
;)

- https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/8352/commits/470ace3d60eb6788fbefdbe8ad658c1e54c6caf9
- https://travis-ci.org/os-autoinst/os-autoinst-distri-opensuse/jobs/579826273

File: tests/installation/grub_test.pm [hunk 1]
```diff
@@ -39,7 +39,9 @@ sub handle_installer_medium_bootup {
         assert_screen 'grub2';
     }
 
-    send_key_until_needlematch 'inst-bootmenu-boot-harddisk', 'up';
+    # Layout of live is different from installation media
+    my $key = is_livecd ? 'down' : 'up';
```
Comment (SergioAtSUSE) [inline tests/installation/grub_test.pm]:
```suggestion
    my $key = is_livecd() ? 'down' : 'up';
```

Comment (ggardet) [reply tests/installation/grub_test.pm]: Updated accordingly. Thanks for your help!

**Pull request merged**